Use ROOT_LIBRARY_PATH in DynamicPath (#7031)
root-project/root: README/ReleaseNotes/v624/index.md

See the discussion at [ROOT-11014](https://sft.its.cern.ch/jira/browse/ROOT-11014) for more context.

### Dynamic Path: `ROOT_LIBRARY_PATH`

> Setting ROOT's "Dynamic Path" gained an additional way of adding to it: the

Reviewer: just wording:

```suggestion
A new way to set ROOT's "Dynamic Path" was added: the
```

?

Author: Included. Should be resolved?

Reviewer: indeed, thanks
